File: mobile/lib/src/models/diary_entry.dart
class DiaryEntry {
  DiaryEntry({
    required this.id,
    required this.userId,
    required this.title,
    required this.entryType,
    required this.entryDate,
    required this.createdAt,
    required this.updatedAt,
    this.content,
    this.mood,
    this.energy,
    this.bestMoment,
    this.challenge,
    this.imageUrl,
    this.imagePublicId,
    this.videoUrl,
    this.videoPublicId,
    this.audioUrl,
    this.audioPublicId,
  });

  final String id;
  final String userId;
  final String title;
  final String? content;
  final String entryType;
  final String? mood;
  final int? energy;
  final String? bestMoment;
  final String? challenge;
  final String? imageUrl;
  final String? imagePublicId;
  final String? videoUrl;
  final String? videoPublicId;
  final String? audioUrl;
  final String? audioPublicId;
  final DateTime entryDate;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory DiaryEntry.fromJson(Map<String, dynamic> json) {
    return DiaryEntry(
      id: json['id'] as String,
      userId: json['user_id'] as String,
      title: json['title'] as String,
      content: json['content'] as String?,
      entryType: json['entry_type'] as String,
      mood: json['mood'] as String?,
      energy: json['energy'] as int?,
      bestMoment: json['best_moment'] as String?,
      challenge: json['challenge'] as String?,
      imageUrl: json['image_url'] as String?,
      imagePublicId: json['image_public_id'] as String?,
      videoUrl: json['video_url'] as String?,
      videoPublicId: json['video_public_id'] as String?,
      audioUrl: json['audio_url'] as String?,
      audioPublicId: json['audio_public_id'] as String?,
      entryDate: DateTime.parse(json['entry_date'] as String),
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'title': title,
      'content': content,
      'entry_type': entryType,
      'mood': mood,
      'energy': energy,
      'best_moment': bestMoment,
      'challenge': challenge,
      'image_url': imageUrl,
      'image_public_id': imagePublicId,
      'video_url': videoUrl,
      'video_public_id': videoPublicId,
      'audio_url': audioUrl,
      'audio_public_id': audioPublicId,
      'entry_date': _dateOnly(entryDate),
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  static String _dateOnly(DateTime date) {
    final year = date.year.toString().padLeft(4, '0');
    final month = date.month.toString().padLeft(2, '0');
    final day = date.day.toString().padLeft(2, '0');
    return '$year-$month-$day';
  }

  String get preview {
    final source = entryType == 'quick'
        ? [bestMoment, challenge].whereType<String>().join(' ')
        : content ?? '';
    if (source.length <= 90) return source;
    return '${source.substring(0, 90)}...';
  }
}

File: backend/app/models/diary_entry.py
import uuid
from datetime import date, datetime
from typing import TYPE_CHECKING

from sqlalchemy import CheckConstraint, Date, DateTime, ForeignKey, Integer, Text, UniqueConstraint, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base

if TYPE_CHECKING:
    from app.models.user import User


class DiaryEntry(Base):
    __tablename__ = "diary_entries"
    __table_args__ = (
        CheckConstraint("entry_type in ('full', 'quick')", name="ck_diary_entries_entry_type"),
        CheckConstraint("energy is null or (energy >= 1 and energy <= 10)", name="ck_diary_entries_energy"),
    )

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), index=True)
    title: Mapped[str] = mapped_column(Text)
    content: Mapped[str | None] = mapped_column(Text)
    entry_type: Mapped[str] = mapped_column(Text)
    mood: Mapped[str | None] = mapped_column(Text)
    energy: Mapped[int | None] = mapped_column(Integer)
    best_moment: Mapped[str | None] = mapped_column(Text)
    challenge: Mapped[str | None] = mapped_column(Text)
    image_url: Mapped[str | None] = mapped_column(Text)
    image_public_id: Mapped[str | None] = mapped_column(Text)
    video_url: Mapped[str | None] = mapped_column(Text)
    video_public_id: Mapped[str | None] = mapped_column(Text)
    audio_url: Mapped[str | None] = mapped_column(Text)
    audio_public_id: Mapped[str | None] = mapped_column(Text)
    entry_date: Mapped[date] = mapped_column(Date)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    user: Mapped["User"] = relationship(back_populates="entries")
    media: Mapped[list["EntryMedia"]] = relationship(
        back_populates="entry",
        cascade="all, delete-orphan",
        order_by="EntryMedia.sort_order",
    )


class EntryMedia(Base):
    __tablename__ = "entry_media"
    __table_args__ = (
        CheckConstraint("media_type in ('image', 'video', 'audio')", name="ck_entry_media_media_type"),
        UniqueConstraint("entry_id", "sort_order", name="uq_entry_media_entry_sort_order"),
    )

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    entry_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("diary_entries.id", ondelete="CASCADE"), index=True)
    media_type: Mapped[str] = mapped_column(Text)
    url: Mapped[str] = mapped_column(Text)
    public_id: Mapped[str | None] = mapped_column(Text)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    entry: Mapped[DiaryEntry] = relationship(back_populates="media")
